src/DocNet/NavigationLevel.cs

```diff
@@ -31,198 +31,325 @@
+                // we have to mark the entry as 'current'
+                if (navigatedPath.Peek() == this && !this.IsRoot)
+                {
+                    fragments.Add("<ul class=\"current\">");
+                }
+                else
+                {
+                    fragments.Add("<ul>");
+                }
+
+                // first render the level header, which is the index element, if present or a label. The root always has an __index element otherwise we'd have stopped at load.
+                var elementStartTag = "<li><span class=\"navigationgroup\"><i class=\"fa fa-caret-down\"></i> ";
+                var indexElement = this.IndexElement;
+                if (indexElement == null)
+                {
+                    fragments.Add(string.Format("{0}{1}</span></li>", elementStartTag, this.Name));
+                }
+                else
+                {
+                    if (this.IsRoot)
+                    {
+                        fragments.Add(indexElement.PerformGenerateToCFragment(navigatedPath, relativePathToRoot));
+                    }
+                    else
+                    {
+                        fragments.Add(string.Format("{0}<a href=\"{1}{2}\">{3}</a></span></li>", elementStartTag, relativePathToRoot, HttpUtility.UrlPathEncode(indexElement.TargetURL),
+                                                    this.Name));
+                    }
+                }
+                // then the elements in the container. Index elements are skipped here.
+                foreach (var element in this.Value)
+                {
+                    fragments.Add(element.GenerateToCFragment(navigatedPath, relativePathToRoot));
+                }
+                fragments.Add("</ul>");
+            }
+            else
+            {
+                // just a link
+                fragments.Add(string.Format("<span class=\"navigationgroup\"><i class=\"fa fa-caret-right\"></i> <a href=\"{0}{1}\">{2}</a></span>",
+                                            relativePathToRoot, HttpUtility.UrlPathEncode(this.TargetURL), this.Name));
+            }
+            if (!this.IsRoot)
+            {
+                fragments.Add("</li>");
+            }
+            return string.Join(Environment.NewLine, fragments.ToArray());
+        }
+
+        private NavigationLevel CreateGeneratedLevel(string path)
+        {
+            var root = new NavigationLevel(_rootDirectory)
+            {
+                ParentContainer = this
+            };
+
+            foreach (var mdFile in Directory.GetFiles(path, "*.md", SearchOption.TopDirectoryOnly))
+            {
+                var name = FindTitleInMdFile(mdFile);
+                if (string.IsNullOrWhiteSpace(name))
+                {
+                    continue;
+                }
+
+                var relativeFilePath = GetRelativePath(_rootDirectory, mdFile);
```

Comment:
There's a MakeRelativePath in Utils.cs in the Docnet project, please use that one.